packages/trpc/server/routers/viewer/teams.tsx

```diff
@@ -35,20 +41,93 @@ export const viewerTeamsRouter = createProtectedRouter()
       teamId: z.number(),
     }),
     async resolve({ ctx, input }) {
-      const team = await getTeamWithMembers(input.teamId);
-      if (!team?.members.find((m) => m.id === ctx.user.id)) {
+      const team = await getTeam(input.teamId);
+      if (!team?.members.find((m) => m.userId === ctx.user.id)) {
         throw new TRPCError({ code: "UNAUTHORIZED", message: "You are not a member of this team." });
       }
-      const membership = team?.members.find((membership) => membership.id === ctx.user.id);
+      const membership = team?.members.find((membership) => membership.userId === ctx.user.id);
 
       return {
         ...team,
         membership: {
           role: membership?.role as MembershipRole,
-          isMissingSeat: membership?.plan === UserPlan.FREE,
           accepted: membership?.accepted,
         },
-        requiresUpgrade: HOSTED_CAL_FEATURES ? !!team.members.find((m) => m.plan !== UserPlan.PRO) : false,
+      };
+    },
+  })
+  .query("getMembers", {
+    input: z.object({
+      teamId: z.number(),
+      limit: z.number().min(1).max(100).nullish(),
+      cursor: z.number().nullish(),
+    }),
+    async resolve({ ctx, input }) {
+      const { teamId, limit, cursor } = input;
+
+      // Check to see that the user is a part of the team
+      const team = await ctx.prisma.team.findFirst({
+        where: {
+          id: teamId,
+        },
+        select: {
+          members: {
+            select: {
+              userId: true,
+            },
+          },
+        },
+      });
+
+      if (!team?.members.some((member) => member.userId !== ctx.user.id)) {
+        throw new TRPCError({ code: "UNAUTHORIZED", message: "You are not a member of this team." });
+      }
```

Comment:
Use this query to confirm that the user making the call is actually a part of the team.